### PR TITLE
Changed homotopyInitialization from parameter to constant

### LINKED ISSUE #1341: change homotopyInitialization to a constant
Change 
```
  parameter Boolean homotopyInitialization = true "= true, use homotopy method"
    annotation(Evaluate=true, Dialog(tab="Advanced"));
```
to a constant with default value `true`, and add a deprecation warning if a user sets it to `false`.

### PR DESCRIPTION
**Pull request by mwetter**
Closes #1341.
This commit also updates the verification script to check for such declarations

### CODE REVIEW

File: IBPSA/Airflow/Multizone/BaseClasses/PowerLawResistance.mo [hunk 1]
```diff
@@ -54,6 +54,10 @@ protected
     "Air mass exchanged (for purpose of error control only)";
 initial equation
   mExc=0;
+  assert(homotopyInitialization, "In " + getInstanceName() +
+    ": The constant homotopyInitialization is set to false. This constant will be removed in future releases.",
```
Comment (Mathadon) [inline IBPSA/Airflow/Multizone/BaseClasses/PowerLawResistance.mo:58]:
```suggestion
    ": The constant homotopyInitialization has been modified from its default value. This constant will be removed in future releases.",
```

File: IBPSA/Fluid/Actuators/BaseClasses/PartialThreeWayValve.mo [hunk 2]
```diff
@@ -52,15 +55,18 @@ partial model PartialThreeWayValve "Partial three way valve"
...
 protected
   Modelica.Blocks.Math.Feedback inv "Inversion of control signal"
     annotation (Placement(transformation(extent={{-74,40},{-62,52}})));
   Modelica.Blocks.Sources.Constant uni(final k=1)
     "Outputs one for bypass valve"
     annotation (Placement(transformation(extent={{-92,40},{-80,52}})));
+
+initial equation
+  assert(homotopyInitialization, "In " + getInstanceName() +
+    ": The constant homotopyInitialization is set to false. This constant will be removed in future releases.",
```
Comment (Mathadon) [inline IBPSA/Fluid/Actuators/BaseClasses/PartialThreeWayValve.mo:67]:
```suggestion
    ": The constant homotopyInitialization has been modified from its default value. This constant will be removed in future releases.",
```

File: IBPSA/Fluid/BaseClasses/PartialResistance.mo [hunk 3]
```diff
@@ -35,6 +37,11 @@ protected
     "Absolute value of nominal flow rate";
   final parameter Modelica.SIunits.PressureDifference dp_nominal_pos(displayUnit="Pa") = abs(dp_nominal)
     "Absolute value of nominal pressure difference";
+initial equation
+  assert(homotopyInitialization, "In " + getInstanceName() +
+    ": The constant homotopyInitialization is set to false. This constant will be removed in future releases.",
```
Comment (Mathadon) [inline IBPSA/Fluid/BaseClasses/PartialResistance.mo:42]:
```suggestion
    ": The constant homotopyInitialization has been modified from its default value. This constant will be removed in future releases.",
```

File: IBPSA/Fluid/Chillers/BaseClasses/Carnot.mo [hunk 4]
```diff
@@ -229,6 +229,9 @@ initial equation
          a=a, x=1)-1) < 0.01, "Efficiency curve is wrong. Need etaPL(y=1)=1.");
   assert(etaCarnot_nominal_internal < 1,   "Parameters lead to etaCarnot_nominal > 1. Check parameters.");
 
+  assert(homotopyInitialization, "In " + getInstanceName() +
+    ": The constant homotopyInitialization is set to false. This constant will be removed in future releases.",
```
Comment (Mathadon) [inline IBPSA/Fluid/Chillers/BaseClasses/Carnot.mo:233]:
```suggestion
    ": The constant homotopyInitialization has been modified from its default value. This constant will be removed in future releases.",
```

File: IBPSA/Fluid/FixedResistances/BaseClasses/PlugFlowCore.mo [hunk 5]
```diff
@@ -134,6 +135,11 @@ protected
     "Default density (e.g., rho_liquidWater = 995, rho_air = 1.2)"
     annotation (Dialog(group="Advanced"));
 
+initial equation
+  assert(homotopyInitialization, "In " + getInstanceName() +
+    ": The constant homotopyInitialization is set to false. This constant will be removed in future releases.",
```
Comment (Mathadon) [inline IBPSA/Fluid/FixedResistances/BaseClasses/PlugFlowCore.mo:140]:
```suggestion
    ": The constant homotopyInitialization has been modified from its default value. This constant will be removed in future releases.",
```

File: IBPSA/Fluid/FixedResistances/Junction.mo [hunk 6]
```diff
@@ -43,8 +46,10 @@ model Junction
     "= true, use linear relation between m_flow and dp for any flow rate"
     annotation(Dialog(tab="Advanced"));
 
-  parameter Boolean homotopyInitialization = true "= true, use homotopy method"
-    annotation(Evaluate=true, Dialog(tab="Advanced"));
+initial equation
+  assert(homotopyInitialization, "In " + getInstanceName() +
+    ": The constant homotopyInitialization is set to false. This constant will be removed in future releases.",
```
Comment (Mathadon) [inline IBPSA/Fluid/FixedResistances/Junction.mo:51]:
```suggestion
    ": The constant homotopyInitialization has been modified from its default value. This constant will be removed in future releases.",
```

File: IBPSA/Fluid/FixedResistances/PlugFlowPipe.mo [hunk 7]
```diff
@@ -151,6 +151,11 @@ protected
     "Default density (e.g., rho_liquidWater = 995, rho_air = 1.2)"
     annotation (Dialog(group="Advanced"));
 
+initial equation
+  assert(homotopyInitialization, "In " + getInstanceName() +
+    ": The constant homotopyInitialization is set to false. This constant will be removed in future releases.",
```
Comment (Mathadon) [inline IBPSA/Fluid/FixedResistances/PlugFlowPipe.mo:156]:
```suggestion
    ": The constant homotopyInitialization has been modified from its default value. This constant will be removed in future releases.",
```

File: IBPSA/Fluid/HeatExchangers/ActiveBeams/BaseClasses/Convector.mo [hunk 8]
```diff
@@ -94,6 +93,12 @@ protected
   Sensors.MassFlowRate senFloWatCoo(
     redeclare final package Medium = Medium) "Mass flow rate sensor"
     annotation (Placement(transformation(extent={{-80,-10},{-60,10}})));
+
+initial equation
+  assert(homotopyInitialization, "In " + getInstanceName() +
+    ": The constant homotopyInitialization is set to false. This constant will be removed in future releases.",
```
Comment (Mathadon) [inline IBPSA/Fluid/HeatExchangers/ActiveBeams/BaseClasses/Convector.mo:99]:
```suggestion
    ": The constant homotopyInitialization has been modified from its default value. This constant will be removed in future releases.",
```

File: IBPSA/Fluid/HeatExchangers/ActiveBeams/Cooling.mo [hunk 9]
```diff
@@ -189,7 +189,9 @@ initial equation
     "Performance curve perCoo.water must pass through (0,0).");
   assert(perCoo.dT.r_dT[1]<=0.000001      and perCoo.dT.f[1]<=0.00001,
     "Performance curve perCoo.dT must pass through (0,0).");
-
+  assert(homotopyInitialization, "In " + getInstanceName() +
+    ": The constant homotopyInitialization is set to false. This constant will be removed in future releases.",
```
Comment (Mathadon) [inline IBPSA/Fluid/HeatExchangers/ActiveBeams/Cooling.mo:193]:
```suggestion
    ": The constant homotopyInitialization has been modified from its default value. This constant will be removed in future releases.",
```

File: IBPSA/Fluid/HeatExchangers/BaseClasses/PartialPrescribedOutlet.mo [hunk 10]
```diff
@@ -38,6 +37,11 @@ protected
     final m_flow_nominal=m_flow_nominal,
     final tau=tau) "Model to set outlet conditions"
     annotation (Placement(transformation(extent={{20,-10},{40,10}})));
+initial equation
+  assert(homotopyInitialization, "In " + getInstanceName() +
+    ": The constant homotopyInitialization is set to false. This constant will be removed in future releases.",
```
Comment (Mathadon) [inline IBPSA/Fluid/HeatExchangers/BaseClasses/PartialPrescribedOutlet.mo:42]:
```suggestion
    ": The constant homotopyInitialization has been modified from its default value. This constant will be removed in future releases.",
```

File: IBPSA/Fluid/HeatExchangers/Radiators/RadiatorEN442_2.mo [hunk 11]
```diff
@@ -218,6 +219,9 @@ initial equation
         x0=0.1*k*(T_b_nominal-TAir_nominal)));
    end for;
 
+  assert(homotopyInitialization, "In " + getInstanceName() +
+    ": The constant homotopyInitialization is set to false. This constant will be removed in future releases.",
```
Comment (Mathadon) [inline IBPSA/Fluid/HeatExchangers/Radiators/RadiatorEN442_2.mo:223]:
```suggestion
    ": The constant homotopyInitialization has been modified from its default value. This constant will be removed in future releases.",
```

File: IBPSA/Fluid/HeatPumps/BaseClasses/PartialWaterToWater.mo [hunk 12]
```diff
@@ -150,6 +151,11 @@ protected
     "Disables compressor when outside of allowed operation range"
     annotation (Placement(transformation(extent={{0,-10},{20,10}})));
 
+initial equation
+  assert(homotopyInitialization, "In " + getInstanceName() +
+    ": The constant homotopyInitialization is set to false. This constant will be removed in future releases.",
```
Comment (Mathadon) [inline IBPSA/Fluid/HeatPumps/BaseClasses/PartialWaterToWater.mo:156]:
```suggestion
    ": The constant homotopyInitialization has been modified from its default value. This constant will be removed in future releases.",
```

File: IBPSA/Fluid/Interfaces/EightPortHeatMassExchanger.mo [hunk 13]
```diff
@@ -324,6 +323,9 @@ initial algorithm
  You need to set massDynamics == Modelica.Fluid.Types.Dynamics.SteadyState to model steady-state.
  Received tau4 = " + String(tau4) + "\n");
 
+  assert(homotopyInitialization, "In " + getInstanceName() +
+    ": The constant homotopyInitialization is set to false. This constant will be removed in future releases.",
```
Comment (Mathadon) [inline IBPSA/Fluid/Interfaces/EightPortHeatMassExchanger.mo:327]:
```suggestion
    ": The constant homotopyInitialization has been modified from its default value. This constant will be removed in future releases.",
```

File: IBPSA/Fluid/Interfaces/FourPortHeatMassExchanger.mo [hunk 14]
```diff
@@ -183,6 +182,9 @@ initial equation
  You need to set massDynamics == Modelica.Fluid.Types.Dynamics.SteadyState to model steady-state.
  Received tau2 = " + String(tau2) + "\n");
 
+  assert(homotopyInitialization, "In " + getInstanceName() +
+    ": The constant homotopyInitialization is set to false. This constant will be removed in future releases.",
```
Comment (Mathadon) [inline IBPSA/Fluid/Interfaces/FourPortHeatMassExchanger.mo:186]:
```suggestion
    ": The constant homotopyInitialization has been modified from its default value. This constant will be removed in future releases.",
```

File: IBPSA/Fluid/Interfaces/StaticFourPortHeatMassExchanger.mo [hunk 15]
```diff
@@ -60,6 +60,12 @@ protected
     final Q_flow = Q2_flow,
     final mWat_flow = mWat2_flow)
     "Model for heat, mass, species, trace substance and pressure balance of stream 2";
+
+initial equation
+  assert(homotopyInitialization, "In " + getInstanceName() +
+    ": The constant homotopyInitialization is set to false. This constant will be removed in future releases.",
```
Comment (Mathadon) [inline IBPSA/Fluid/Interfaces/StaticFourPortHeatMassExchanger.mo:66]:
```suggestion
    ": The constant homotopyInitialization has been modified from its default value. This constant will be removed in future releases.",
```

File: IBPSA/Fluid/Interfaces/StaticTwoPortHeatMassExchanger.mo [hunk 16]
```diff
@@ -59,6 +59,12 @@ protected
   Modelica.Blocks.Sources.RealExpression
     masExc(final y=mWat_flow) "Block to set moisture exchange in volume"
     annotation (Placement(transformation(extent={{-20,20},{0,40}})));
+
+initial equation
+  assert(homotopyInitialization, "In " + getInstanceName() +
+    ": The constant homotopyInitialization is set to false. This constant will be removed in future releases.",
```
Comment (Mathadon) [inline IBPSA/Fluid/Interfaces/StaticTwoPortHeatMassExchanger.mo:65]:
```suggestion
    ": The constant homotopyInitialization has been modified from its default value. This constant will be removed in future releases.",
```

File: IBPSA/Fluid/Interfaces/TwoPortHeatMassExchanger.mo [hunk 17]
```diff
@@ -89,6 +88,9 @@ initial algorithm
  You need to set massDynamics == Modelica.Fluid.Types.Dynamics.SteadyState to model steady-state.
  Received tau = " + String(tau) + "\n");
 
+  assert(homotopyInitialization, "In " + getInstanceName() +
+    ": The constant homotopyInitialization is set to false. This constant will be removed in future releases.",
```
Comment (Mathadon) [inline IBPSA/Fluid/Interfaces/TwoPortHeatMassExchanger.mo:92]:
```suggestion
    ": The constant homotopyInitialization has been modified from its default value. This constant will be removed in future releases.",
```

File: IBPSA/Fluid/Movers/BaseClasses/FlowMachineInterface.mo [hunk 18]
```diff
@@ -337,6 +337,10 @@ the simulation stops.");
      else IBPSA.Utilities.Math.Functions.splineDerivatives(x=per.hydraulicEfficiency.V_flow,
     y=per.hydraulicEfficiency.eta);
 
+  assert(homotopyInitialization, "In " + getInstanceName() +
+    ": The constant homotopyInitialization is set to false. This constant will be removed in future releases.",
```
Comment (Mathadon) [inline IBPSA/Fluid/Movers/BaseClasses/FlowMachineInterface.mo:341]:
```suggestion
    ": The constant homotopyInitialization has been modified from its default value. This constant will be removed in future releases.",
```

File: IBPSA/Fluid/Movers/BaseClasses/PowerInterface.mo [hunk 19]
```diff
@@ -47,6 +47,11 @@ protected
   Modelica.SIunits.HeatFlowRate QThe_flow
     "Heat input from fan or pump to medium";
 
+initial equation
+  assert(homotopyInitialization, "In " + getInstanceName() +
+    ": The constant homotopyInitialization is set to false. This constant will be removed in future releases.",
```
Comment (Mathadon) [inline IBPSA/Fluid/Movers/BaseClasses/PowerInterface.mo:52]:
```suggestion
    ": The constant homotopyInitialization has been modified from its default value. This constant will be removed in future releases.",
```

File: IBPSA/Fluid/Storage/BaseClasses/IndirectTankHeatExchanger.mo [hunk 20]
```diff
@@ -152,6 +152,11 @@ protected
         rotation=90,
         origin={20,42})));
 
+initial equation
+  assert(homotopyInitialization, "In " + getInstanceName() +
+    ": The constant homotopyInitialization is set to false. This constant will be removed in future releases.",
```
Comment (Mathadon) [inline IBPSA/Fluid/Storage/BaseClasses/IndirectTankHeatExchanger.mo:157]:
```suggestion
    ": The constant homotopyInitialization has been modified from its default value. This constant will be removed in future releases.",
```

**Comment by mwetter** [comment]
@Mathadon : Thanks for the reformulation which I added in https://github.com/ibpsa/modelica-ibpsa/pull/1342/commits/dbb8586633fe82a743e7a8baac24559b35e9956f .
 This is ready to merge when all tests pass.

**Pull request merged**